components/slidingPanel/slidingPanel.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component, PropTypes } from 'react';
+import { findDOMNode } from 'react-dom';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+
+export default class SlidingPanel extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { hideOverlay: true, isOpen: false };
+
+    this.handleClickOverlay = this.handleClickOverlay.bind(this);
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleTransitionEnd = this.handleTransitionEnd.bind(this);
+  }
+
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.node),
+      PropTypes.node,
+    ]).isRequired,
+
+    /**
+     * When true, if the user clicks on the overaly, closes the panel.
+     */
+    closeOnOverlay: PropTypes.bool,
+
+    /**
+     * Data attributes. You can use it to set up any custom data-* attribute
+     */
+    dataAttrs: PropTypes.object,
+
+    /**
+     * You can provide set of custom modifications.
+     */
+    mods: PropTypes.arrayOf(PropTypes.string),
+
+    /**
+     * When defined, this function is triggered when the panel is closing.
+     */
+    onClose: PropTypes.func,
+
+    /**
+     * When defined, this function is triggered when the panel is opening.
+     */
+    onOpen: PropTypes.func,
+
+    /**
+     * Defines if the panel is open.
+     */
+    open: PropTypes.bool,
```

Comment:
I agree that the name is bad, as I mentioned before, but I think it's way worst if we have components accept different prop naming for the same use cases. If it were that easy, I would rename the `active` prop in Modal, but it's already in use.